crates/iota-sdk-ffi/src/crypto/multisig.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    borrow::Cow,
+    sync::{Arc, RwLock},
+};
+
+use iota_crypto::{SignatureError, Verifier};
+
+use crate::{
+    crypto::zklogin::ZkloginVerifier,
+    error::Result,
+    types::{
+        crypto::multisig::{MultisigAggregatedSignature, MultisigCommittee},
+        signature::UserSignature,
+        transaction::Transaction,
+    },
+};
+
+#[derive(derive_more::From, uniffi::Object)]
+pub struct MultisigVerifier(pub iota_crypto::multisig::MultisigVerifier);
+
+#[uniffi::export]
+impl MultisigVerifier {
+    #[uniffi::constructor]
+    pub fn new() -> Self {
+        Self(iota_crypto::multisig::MultisigVerifier::new())
+    }
+
+    pub fn with_zklogin_verifier(&self, zklogin_verifier: &ZkloginVerifier) -> Self {
+        let mut verifier = self.0.clone();
+        verifier.with_zklogin_verifier(zklogin_verifier.0.clone());
+        Self(verifier)
+    }
+
+    pub fn zklogin_verifier(&self) -> Option<Arc<ZkloginVerifier>> {
+        self.0
+            .zklogin_verifier()
+            .cloned()
+            .map(Into::into)
+            .map(Arc::new)
+    }
+
+    pub fn verify(&self, message: &[u8], signature: &MultisigAggregatedSignature) -> Result<()> {
```

Comment:
Maybe add the `_user` version for ease of use?

Reply:
I honestly don't see the point. All it does is call `.as_multisig()` for you before passing it in. I dunno